cmux.go

```diff
@@ -163,11 +172,17 @@ func (m *cMux) serve(c net.Conn, donec <-chan struct{}, wg *sync.WaitGroup) {
 	defer wg.Done()
 
 	muc := newMuxConn(c)
+	if m.matchTimeout > 0 {
+		_ = c.SetReadDeadline(time.Now().Add(m.matchTimeout))
+	}
 	for _, sl := range m.sls {
 		for _, s := range sl.ss {
 			matched := s(muc.Conn, muc.startSniffing())
 			if matched {
 				muc.doneSniffing()
+				if m.matchTimeout > 0 {
+					_ = c.SetReadDeadline(time.Time{})
```

Comment:
Please define a `noDeadline` var for readability similar to: https://golang.org/src/net/net.go